### PR TITLE
Meine Republik Teaser

### PR DESCRIPTION
**Pull request by ovbm**
Screenshot 2020-09-08 at 09 15 32

### CODE REVIEW

File: src/components/TeaserActiveDebates/ActiveDebates.js [hunk 1]
```diff
@@ -207,79 +207,62 @@ ActiveDebates.data = {
     }
   },
   query: `
-query getFrontDiscussions($lastDays: Int!, $first: Int!, $featured: Int!) {
```
Comment (ovbm) [inline src/components/TeaserActiveDebates/ActiveDebates.js:210]: why this file?

**Comment by ovbm** [comment]
@tpreusse I think this is ready for review.

File: src/components/TeaserMyMagazine/TeaserMyMagazine.js [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    documents: search(
+      filters: [
+          { key: "template", not: true, value: "section" }
+          { key: "template", not: true, value: "format" }
+          { key: "template", not: true, value: "front" }
+        ]
+        filter: { feed: true }
+        sort: { key: publishedAt, direction: DESC }
+        first: 2
+      ) {
```
Comment (tpreusse) [inline src/components/TeaserMyMagazine/TeaserMyMagazine.js:214]: This query needs to change to get personalised documents. Something like:
```gql
{
  notifications(first: 2, filter: Document) {
    nodes {
      object {
        ... on Document {
          id
          meta {
            title
          }
        }
      }
    }
  }
}
```

Needs a backend change to allow to get only document notifications. I created a new filter for it in the backend: https://github.com/orbiting/backends/pull/514

File: src/templates/Front/liveTeasers.js [hunk 3]
```diff
@@ -34,8 +35,62 @@ const createLiveTeasers = ({
...
+  ActionBar,
+  showMyMagazine = true
 }) => {
+  const MyMagazineWithData = withMyMagazineData(
+    ({ attributes, data, url, label }) => {
+      return (
+        <Loader
+          error={data.error}
+          loading={data.loading}
+          style={{ minHeight: 600 }}
```
Comment (tpreusse) [inline src/templates/Front/liveTeasers.js:49]: I'd recommend to move the `600` to a shared variable to ensure that the `Loader` and `LazyLoad` placeholder always have the same size.

**Review by tpreusse — COMMENTED** [review]
General approval but one more thing re text color—use the rule provided by the context.

Needs testing on staging.

File: src/components/TeaserFeed/Credit.js [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
-    },
-    color: colors.text
+    }
   })
 }
 
 const Credit = ({ children }) => {
-  return <p {...styles.main}>{children}</p>
+  const [colorScheme] = useColorContext()
+  return <p {...css(styles.main, { color: colorScheme.text })}>{children}</p>
```
[the remark this answers is not in the document]
Comment (ovbm) [reply src/components/TeaserFeed/Credit.js]: done

File: src/components/TeaserMyMagazine/TeaserMyMagazine.js [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+        nodes {
+          object {
+            ... on Document {
+              id
+              meta {
+                shortTitle
+                title
+                credits
+                format {
+                  meta {
```
Comment (tpreusse) [inline src/components/TeaserMyMagazine/TeaserMyMagazine.js:235]:
```suggestion
                format {
                  id
                  meta {
```

File: src/components/TeaserMyMagazine/TeaserMyMagazine.js [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+                    path
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+      me {
+        bookmarkAndProgress: collectionItems(names: ["progress", "bookmarks"], first: 2) {
```
Comment (tpreusse) [inline src/components/TeaserMyMagazine/TeaserMyMagazine.js:248]:
```suggestion
      me {
        id
        bookmarkAndProgress: collectionItems(names: ["progress", "bookmarks"], first: 2) {
```

**Review by tpreusse — COMMENTED** [review]
I think I found the `shortTitle` issue: it not defined in the `latestProgressOrBookmarkedArticles` loop.

File: src/components/TeaserMyMagazine/TeaserMyMagazine.js [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+              <div {...styles.sectionTitle}>
+                <Link href={notificationsUrl} passHref>
+                  <TeaserSectionTitle small href={notificationsUrl}>
+                    {notificationsLabel}
+                  </TeaserSectionTitle>
+                </Link>
+              </div>
+              {latestProgressOrBookmarkedArticles.map(document => {
+                const { id } = document
+                const { path } = document.meta
```
[the remark this answers is not in the document]
Comment (tpreusse) [reply src/components/TeaserMyMagazine/TeaserMyMagazine.js:53]: also note: `title` was also undefined. Which is also the reason the styleguide deploys failed:
```
       Failed to compile.
       
       ./src/components/TeaserMyMagazine/TeaserMyMagazine.js
         Line 66:26:  'shortTitle' is not defined  no-undef
         Line 66:40:  'shortTitle' is not defined  no-undef
         Line 67:32:  'shortTitle' is not defined  no-undef
         Line 68:31:  'shortTitle' is not defined  no-undef
         Line 69:32:  'title' is not defined       no-undef
         Line 70:33:  'title' is not defined       no-undef
```

[the remark this answers is not in the document]
Comment (ovbm) [reply src/components/TeaserMyMagazine/TeaserMyMagazine.js:53]: i think that’s not it, but an error when i removed certain code blocks and forgot to put it back

**Comment by ovbm** [comment]
@tpreusse würde das jetzt mergen.

File: src/components/TeaserMyMagazine/TeaserMyMagazine.js [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+  latestSubscribedArticles: PropTypes.array,
+  latestProgressOrBookmarkedArticles: PropTypes.array
+}
+
+WrappedTeaserMyMagazine.data = {
+  config: {
+    options: ({ first = 2 }) => ({
+      variables: {
+        first: +first
+      }
```
Comment (tpreusse) [inline src/components/TeaserMyMagazine/TeaserMyMagazine.js:223]:
```suggestion
      variables: {
        first: +first
      },
      ssr: false
```
And could you please add the same to the options of `TeaserActiveDebates/ActiveDebates.js?

**Review by tpreusse — APPROVED** [review]
One last thing: let's turn off SSR explicitly.

**Pull request merged**